src/snowflake/connector/network_async.py

```diff
@@ -178,6 +224,15 @@ def encode(self) -> str:
 
 
 class SessionPoolAsync(SessionPool):
+    def get_session_async(self) -> aiohttp.ClientSession:
+        """Returns a session from the session pool or creates a new one."""
+        try:
+            session = self._idle_sessions.pop()
+        except IndexError:
+            session = self._rest.make_requests_session_async()
+        self._active_sessions.add(session)
```

Comment:
is there a limitation on the number of sessions being created? how do we do this in the sync one

Reply:
I don't believe there is a limitation in our sync code either.

Reply:
There isn't